modules/python/tests/test_python_persist.c

```diff
@@ -180,7 +180,7 @@ persist = Persist('persist_name')\n\
 persist['integer'] = 1\n\
 persist['str'] = 'str'\n\
 persist['bytes'] = b'bytes'\n\
-assert sorted([1, 'str', b'bytes']) == sorted([persist[k] for k in persist])";
```

Comment:
Can you give a brief explanation about the reason of changing `sorted` to `set`?
In my understanding the difference in this case is that `set` removes the duplicated elements. Is that expected?

Reply:
The reason is: in python3, `sorted` does not work on lists that have both integer and string arguments. Originally, sorted was needed because the right side is a hashtable, and it can return the entries in any order. In python2, `sorted` worked. With using `set`, I can compare the two objects easily.

The new version does not detect bugs when values and keys are shuffled. We partially defend against this in another testcase:
https://github.com/syslog-ng/syslog-ng/blob/3a250f4c2dbcd5554760166ded86542459ca2acb/modules/python/tests/test_python_persist.c#L154
It also does not defend against if the left side has an unknown key, which overlaps with a known value. I was thinking this test is good enough when trading complexity, but of course, using `sorted` with a moderately complex key function: for example map to `[str(type(v)), str(v))]`, this can be resolved, if you think it would worth doing so. Or may be there are easier alternatives.

Reply:
We can control the correct inputs for this test, so this is fine, I think.

Thanks for the explanation :)